include/seqan3/range/view/pairwise_combine.hpp

```diff
@@ -0,0 +1,329 @@
+ * while the order of selection does not matter. A source range with `n` elements will therefore have `n choose 2`
+ * possible combinations.
+ */
+template <std::ranges::View source_range_type>
+//!\cond
+    requires  std::ranges::ForwardRange<source_range_type>
+// !\endcond
+class pairwise_combine_view : public ranges::view_interface<pairwise_combine_view<source_range_type>>
+{
+private:
+
+    //!\brief Alias type for the iterator over the source range.
+    using source_iterator_type = std::ranges::iterator_t<source_range_type>;
+
+    /*!\brief The internal iterator type.
+     *
+     * \details
+     *
+     * This iterator models the iterator category of the `source_iterator_type`. It maintains a pair of iterators on
+     * the source range that move over all pairwise combinations of elements. The end is reached, when the second
+     * iterator points to the end of the source range and the first iterator to the last element of the range.
+     */
+    class iterator_type
+    {
+        //!\brief Alias for the value type of the source iterator type.
+        using source_val_t = typename std::iterator_traits<source_iterator_type>::value_type;
+        //!\brief Alias for the reference type of the source iterator type.
+        using source_ref_t = typename std::iterator_traits<source_iterator_type>::reference;
+    public:
+
+        /*!\name Associated types
+         * \{
+         */
+        using difference_type       = std::ptrdiff_t;
+        using value_type            = std::tuple<source_val_t, source_val_t>;
+        using reference             = std::tuple<source_ref_t, source_ref_t>;
+        using pointer               = void;
+        using iterator_category     = typename std::iterator_traits<source_iterator_type>::iterator_category;
+        //!\}
+
+        /*!\name Constructors, destructor and assignment
+         * \{
+         */
+        constexpr iterator_type() = default;
+        constexpr iterator_type(iterator_type const &) = default;
+        constexpr iterator_type(iterator_type &&) = default;
+        constexpr iterator_type & operator=(iterator_type const &) = default;
+        constexpr iterator_type & operator=(iterator_type &&) = default;
+        ~iterator_type() = default;
+
+        /*!\brief Constructs the iterator from the current source iterator and the end iterator of the source range.
+         * \param[in] iter   The iterator pointing to current element within the source range.
+         * \param[in] end_it The iterator pointing to end of the source range.
+         *
+         * \details
+         *
+         * Constructs the iterator by caching the end of the source range and setting the first iterator to the
+         * given position and the second to the first incremented by one.
+         */
+        constexpr iterator_type(source_iterator_type iter,
+                                source_iterator_type end_it) noexcept :
```

Comment:
Pff you thumbed :+1: my review request but the indentation is still weird...

Reply:
Everything seems perfectly aligned to me? I just wrapped each initialiser on its own line. Since when is this weird?

Reply:
well now that its three parameters that don't fit in one line it's fine, but if function parameters fit into one line we usually put them in one line.. and I never saw the parameter initialisation in multiple lines either when they fit into one line. But I guess that's not a style guide requirement anyway so sorry for the request.

Reply:
No it is ok, but I always put them on single line once they cannot be put immediately behind the colon. But as you said we don’t have a guide for this